### PR TITLE
Pipeline catalog integration

### PR DESCRIPTION
**Pull request by gitlawr**
Address issue: https://github.com/rancher/rancher/issues/16026
Depending on types pr: https://github.com/rancher/types/pull/630

A project system user with project-member rolebinding is created for a project, so as to call app related APIs in the build pods.

### CODE REVIEW

File: pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go [hunk 1]
```diff
@@ -61,6 +61,15 @@ func (l *Lifecycle) deploy(obj *v3.PipelineExecution) error {
 		return errors.Wrapf(err, "Error create secret")
 	}
 
+	apikey, err := l.systemAccountManager.GetOrCreateProjectSystemToken(projectID)
+	if err != nil {
+		return err
+	}
+	secret = getAPIKeySecret(nsName, apikey)
+	if _, err := l.secrets.Create(secret); err != nil && !apierrors.IsAlreadyExists(err) {
```
Comment (orangedeng) [inline pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go]: Do we create secret for each deploy? Should we recycle them?

Comment (gitlawr) [reply pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go]: Its lifecyle is the same as pipeline components.

Comment (StrongMonkey) [reply pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go]: I think you should use lister to get secret from cache first, if it doesn't exist, then create it.

Comment (gitlawr) [reply pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go]: Probably better if we find the dedicated namespace exists at the begining, we skip the whole deploy stuff. Any missing part would be addressed in upgrade logic.

Comment (gitlawr) [reply pkg/controllers/user/pipeline/controller/pipelineexecution/deploy.go]: Updated

**Comment by orangedeng** [comment]
LGTM

**Comment by StrongMonkey** [comment]
LGTM

**Pull request merged**